testsuite/MDAnalysisTests/coordinates/test_netcdf.py

```diff
@@ -178,23 +179,51 @@ def tearDown(self):
             except OSError:
                 pass
         del self.universe
-        del self.Writer
         del self.tmpdir
 
-    def test_write_trajectory(self):
+    def _test_write_trajectory(self):
+        # explicit import so that we can artifically remove netCDF4
+        # before calling
+        from MDAnalysis.coordinates import TRJ
+
         t = self.universe.trajectory
-        with self.Writer(self.outfile, t.n_atoms, dt=t.dt) as W:
+        with TRJ.NCDFWriter(self.outfile, t.n_atoms, dt=t.dt) as W:
             self._copy_traj(W)
         self._check_new_traj()
-        import netCDF4
         #for issue #518 -- preserve float32 data in ncdf output
-        dataset = netCDF4.Dataset(self.outfile, 'r', format='NETCDF3')
+        # NOTE: This originally failed with the dtype('>f4') instead
+        #       of dtype('<f4') == dtype('f') == np.float32, i.e. then
+        #       endianness is different. The current hack-ish solution
+        #       ignores endianness by comparing the name of the types,
+        #       which should be "float32".
+        #       See http://docs.scipy.org/doc/numpy-1.10.0/reference/arrays.dtypes.html
+        #       and https://github.com/MDAnalysis/mdanalysis/pull/503
+        dataset = netcdf.netcdf_file(self.outfile, 'r')
         coords = dataset.variables['coordinates']
         time = dataset.variables['time']
-        assert_equal(coords.dtype, np.float32,
-                     err_msg='ncdf coord output not float32')
-        assert_equal(time.dtype, np.float32,
-                     err_msg='ncdf time output not float32')
+        assert_equal(coords[:].dtype.name, np.dtype(np.float32).name,
+                     err_msg='ncdf coord output not float32 '
+                             'but {}'.format(coords[:].dtype))
+        assert_equal(time[:].dtype.name, np.dtype(np.float32).name,
+                err_msg='ncdf time output not float32 '
+                        'but {}'.format(time[:].dtype))
+
+    def test_write_trajectory_netCDF4(self):
+        pytest.importorskip("netCDF4")
+        return self._test_write_trajectory()
+
+    def test_write_trajectory_netcdf(self):
+        import MDAnalysis.coordinates.TRJ
+        loaded_netCDF4 = sys.modules['MDAnalysis.coordinates.TRJ'].netCDF4
+        try:
+            # cannot use @block_import('netCDF4') because TRJ was already imported
```

Comment:
Sorry for not checking this sooner, this seems correct.  `block_import` will sniff and block `import` calls, but it's too late once it's already imported.